moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -852,6 +857,54 @@ void ompl_interface::ModelBasedPlanningContext::unregisterTerminationCondition()
   ptc_ = nullptr;
 }
 
+int32_t ompl_interface::ModelBasedPlanningContext::errorCode(const ompl::base::PlannerStatus& status)
+{
+  auto result = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
+  switch (ompl::base::PlannerStatus::StatusType(status))
+  {
+    case ompl::base::PlannerStatus::UNKNOWN:
+      ROS_WARN_NAMED(LOGNAME, "Motion planning failed for an unknown reason");
+      result = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
+      break;
+    case ompl::base::PlannerStatus::INVALID_START:
+      ROS_WARN_NAMED(LOGNAME, "Invalid start state");
+      result = moveit_msgs::MoveItErrorCodes::START_STATE_INVALID;
+      break;
+    case ompl::base::PlannerStatus::INVALID_GOAL:
+      ROS_WARN_NAMED(LOGNAME, "Invalid goal state");
+      result = moveit_msgs::MoveItErrorCodes::GOAL_STATE_INVALID;
+      break;
+    case ompl::base::PlannerStatus::UNRECOGNIZED_GOAL_TYPE:
+      ROS_WARN_NAMED(LOGNAME, "Unrecognized goal type");
+      result = moveit_msgs::MoveItErrorCodes::UNRECOGNIZED_GOAL_TYPE;
+      break;
+    case ompl::base::PlannerStatus::TIMEOUT:
+      ROS_WARN_NAMED(LOGNAME, "Timed out");
+      result = moveit_msgs::MoveItErrorCodes::TIMED_OUT;
+      break;
+    case ompl::base::PlannerStatus::APPROXIMATE_SOLUTION:
+      ROS_WARN_NAMED(LOGNAME, "Solution is approximate");
+      result = moveit_msgs::MoveItErrorCodes::SUCCESS;
```

Comment:
```suggestion
      result = moveit_msgs::MoveItErrorCodes::PLANNING_FAILED;
```
"approximate" can be anything really, for RRTConnect it can refer to a "solution" that consists of only a couple of the steps near the goal

Reply:
I looked it up again because I didn't remember the details. Officially `approximate` means "any trajectory that starts at the start state but does not reach the goal state". There was a bug in RRTConnect where the opposite would also be approximate but that was fixed in https://github.com/ompl/ompl/pull/739    ... which has not been released yet

Reply:
Yes, especially with the bug in OMPL still present in release builds this could give you "solutions" that contain a huge jump towards the goal state. Once this is released we could reconsider if a solution that "gets you away from start" can be somehow useful in any case? Note that it would not necessarily be closer to the goal or the goal easier to reach from the new position ...

Reply:
Done -- thanks